pkg/cvo/upgradeable.go

```diff
@@ -2,34 +2,50 @@ package cvo
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
 	"strings"
 	"time"
 
 	configv1 "github.com/openshift/api/config/v1"
 	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
 	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	listerscorev1 "k8s.io/client-go/listers/core/v1"
 	"k8s.io/klog/v2"
 
 	"github.com/openshift/cluster-version-operator/lib/resourcedelete"
 	"github.com/openshift/cluster-version-operator/lib/resourcemerge"
+	"github.com/openshift/cluster-version-operator/pkg/internal"
+	"github.com/openshift/cluster-version-operator/pkg/payload/precondition/clusterversion"
 )
 
+const adminAckGateFmt string = "^ack-[4-5]\\.([0-9]{1,})"
```

Comment:
nit: you can use backticked [raw string literals][1] to avoid the doubled escape:

```go
const adminAckGateFmt string = `^ack-[4-5]\.([0-9]{1,})`
```

Although I often just use a range:

```go
const adminAckGateFmt string = "^ack-[4-5][.]([0-9]{1,})"
```

[1]: https://golang.org/ref/spec#String_literals

Reply:
Also, if this is only feeding the `MustCompile`, we can probably drop this `const` and inline:

```go
var adminAckGateRegexp = regexp.MustCompile("^ack-[4-5][.]([0-9]{1,})")
```

Reply:
Hmm, also, should the regexp include a trailing `-`?  `^ack-[4-5][.]([0-9]{1,})-`

Reply:
changed